src/medCore/medPlugin.h

```diff
@@ -0,0 +1,96 @@
+/*=========================================================================
+
+ medInria
+
+ Copyright (c) INRIA 2013 - 2014. All rights reserved.
+ See LICENSE.txt for details.
+
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#pragma once
+
+#include <dtkLog/dtkLog.h>
+#include <dtkCore/dtkPlugin.h>
+#include <medCoreExport.h>
+
+class MEDCORE_EXPORT medPlugin : public dtkPlugin
+{
+    Q_OBJECT
+    Q_INTERFACES(dtkPlugin)
+
+public:
+    medPlugin(QObject* parent = 0);
+
+    /**
+     * @brief name() fills the header widget in the "i" information pop-up.
+     *        Is also used to link a toolbox with a plugin in its plugin() function
+     * @return the name of the plugin
+     */
+    virtual QString name() const = 0;
```

Comment:
this is the only function absolutely needed by a plugin. The others are not always found in plugins